keras_cv/models/backbones/resnet_v1/resnet_v1_backbone_test.py

```diff
@@ -0,0 +1,190 @@
+        model = ResNetBackbone(
+            stackwise_filters=[64, 128, 256, 512],
+            stackwise_blocks=[2, 2, 2, 2],
+            stackwise_strides=[1, 2, 2, 2],
+            include_rescaling=False,
+        )
+        model_2 = ResNetBackbone(
+            stackwise_filters=[64, 128, 256, 512],
+            stackwise_blocks=[2, 2, 2, 2],
+            stackwise_strides=[1, 2, 2, 2],
+            include_rescaling=False,
+        )
+        layers_1 = model.layers
+        layers_2 = model_2.layers
+        for i in range(len(layers_1)):
+            if "input" in layers_1[i].name:
+                continue
+            self.assertEquals(layers_1[i].name, layers_2[i].name)
+
+    def test_create_backbone_model_from_alias_model(self):
+        model = ResNet50Backbone(
+            include_rescaling=False,
+        )
+        backbone_model = get_feature_extractor(
+            model,
+            model.pyramid_level_inputs.values(),
+            model.pyramid_level_inputs.keys(),
+        )
+        inputs = tf.keras.Input(shape=[256, 256, 3])
+        outputs = backbone_model(inputs)
+        # Resnet50 backbone has 4 level of features (2 ~ 5)
+        self.assertLen(outputs, 4)
+        self.assertEquals(list(outputs.keys()), [2, 3, 4, 5])
+        self.assertEquals(outputs[2].shape, [None, 64, 64, 256])
+        self.assertEquals(outputs[3].shape, [None, 32, 32, 512])
+        self.assertEquals(outputs[4].shape, [None, 16, 16, 1024])
+        self.assertEquals(outputs[5].shape, [None, 8, 8, 2048])
+
+    def test_create_backbone_model_with_level_config(self):
+        model = ResNetBackbone(
+            stackwise_filters=[64, 128, 256, 512],
+            stackwise_blocks=[2, 2, 2, 2],
+            stackwise_strides=[1, 2, 2, 2],
+            include_rescaling=False,
+            input_shape=[256, 256, 3],
+        )
+        levels = [3, 4]
+        layer_names = [model.pyramid_level_inputs[level] for level in [3, 4]]
+        backbone_model = get_feature_extractor(model, layer_names, levels)
+        inputs = tf.keras.Input(shape=[256, 256, 3])
+        outputs = backbone_model(inputs)
+        self.assertLen(outputs, 2)
+        self.assertEquals(list(outputs.keys()), [3, 4])
+        self.assertEquals(outputs[3].shape, [None, 32, 32, 512])
+        self.assertEquals(outputs[4].shape, [None, 16, 16, 1024])
+
+    @parameterized.named_parameters(
+        ("one_channel", 1),
+        ("four_channels", 4),
+    )
+    def test_application_variable_input_channels(self, num_channels):
+        # ResNet50 model
+        model = ResNetBackbone(
+            stackwise_filters=[64, 128, 256, 512],
+            stackwise_blocks=[3, 4, 6, 3],
+            stackwise_strides=[1, 2, 2, 2],
+            input_shape=(None, None, num_channels),
+            include_rescaling=False,
+        )
+        self.assertEqual(model.output_shape, (None, None, None, 2048))
+
+    @parameterized.named_parameters(
```

Comment:
What is this testing? Should we at least check the forward pass?

Reply:
Done.

Reply:
Maybe add these to ResNetV2 as well for consistency.